common/utils.py

```diff
@@ -130,10 +132,22 @@ def origin(string):
     return 'open-humans' if string == 'open-humans' else 'external'
 
 
-def get_redirect_url(request):
-    redirect_to = get_request_param(self.request, 'next')
-    if redirect_to:
-        url = is_safe_url(unquote_plus(redirect_to), allowed_hosts=None)
+def strip_redirect_url(raw_url):
+    """
+    The reset password and signup links are, well, links.  Thus, they do not
```

Comment:
Not sure I quite get what's going on here from this docstring.

Reply:
This docstring still confuses me – I really don't understand what it's saying.

Actually, and maybe related: I think I asked if I could get a diagram mapping out the logic actually, like when next parameters are used (and precedence). I still think this would help me understand how the new redirect methods are working.

Reply:
My wordy attempt.

In the case of hitting a "login needed" page where PrivateMixin is present, that page is stored in session. Otherwise, the user has the options to do (a) password-based login/signup, which when POSTed, checks HTTP_REFERER. Because these are javascript pop-up forms, that URL is the "original location" the user expects to return to. This is stored in session (b) social login/signup, which sends a user to a view which (passes them to a provider, but first) records the referring URL (HTTP_REFERER) in the GET request into session.